src/auth/index.js

```diff
@@ -0,0 +1,127 @@
+import Vue from 'vue'
+import createAuth0Client from '@auth0/auth0-spa-js'
+import { store } from '@/store'
+import { mapActions } from 'vuex'
+import {
+  FETCH_USER_SUBSCRIPTION
+} from '@/store/action-types'
+
+// fix for window.crypto is required bug. See #522
+export const createAuthLink = async () => await createAuth0Client({
+  domain: process.env.VUE_APP_AUTH0_DOMAIN,
+  client_id: process.env.VUE_APP_AUTH0_CLIENTID,
+  redirect_uri: process.env.VUE_APP_REDIRECT_URL
+})
+
+const DEFAULT_REDIRECT_CALLBACK = () =>
+  window.history.replaceState({}, document.title, window.location.pathname)
+
+let instance
+
+export const useAuth0 = ({
+  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
+  redirectUri = window.location.origin,
+  user = {},
+  ...options
+}) => {
+  if (instance) return instance
+
+  instance = new Vue({
+    store,
+    data: () => ({
+      auth0Client: null
+    }),
+    methods: {
+      ...mapActions([FETCH_USER_SUBSCRIPTION]),
+      async handleRedirectCallback () {
+        this.loading = true
+        try {
+          await this.auth0Client.handleRedirectCallback()
+
+          const user = await this.auth0Client.getUser()
+          this.$store.dispatch('SAVE_AUTHENTICATED_USER', {
+            user,
+            loadingAuth: false,
+            isAuthenticated: true
+          })
+
+          this.$store.dispatch(FETCH_USER_SUBSCRIPTION, user.sub)
+        } catch (e) {
+          this.error = e
+
+          this.$store.dispatch('SAVE_AUTHENTICATED_USER', {
+            user: null,
+            loadingAuth: false,
+            isAuthenticated: false
+          })
+        } finally {
+          this.loading = false
+        }
+      },
+      loginWithRedirect (o) {
+        return this.auth0Client.loginWithRedirect(o)
+      },
+      logout (o) {
+        this.$store.dispatch('SAVE_AUTHENTICATED_USER', {
+          user: null,
+          loadingAuth: false,
+          isAuthenticated: false
+        })
+
+        return this.auth0Client.logout(o)
+      },
+      async authenticateUser () {
+        try {
+          if (
+            window.location.search.includes('code=') &&
+              window.location.search.includes('state=')
+          ) {
+            const { appState } = await this.auth0Client.handleRedirectCallback()
+            this.error = null
+            onRedirectCallback(appState)
+          }
+        } catch (e) {
+          this.error = e
+        } finally {
+          if (this.auth0Client) {
+            this.isAuthenticated = await this.auth0Client.isAuthenticated()
+
+            const user = await this.auth0Client.getUser()
+
+            this.$store.dispatch('SAVE_AUTHENTICATED_USER', {
```

Comment:
@vickywane similar issue here as before. Because the code does not `await` for async call nor does it check the result of returned promises, there is no easy way of knowing what the execution order is, nor what the outcome is from these calls. Makes it very hard to debug issues.

Reply:
@ivelin [This](https://github.com/vickywane/ambianic-ui/blob/feature/user-management/package.json#L14) is the link to the script that combines the coverage files from jest and cypress. [This](https://github.com/vickywane/ambianic-ui/blob/feature/user-management/package.json#L14) is also the command within the `workflow` file to generate the reports within the workflow build.

It was originally implemented in [this](https://github.com/bahmutov/cypress-and-jest/blob/master/package.json) project.